azurerm/validators_test.go

```diff
@@ -85,3 +89,50 @@ func TestValidateIntInSlice(t *testing.T) {
 	}
 
 }
+
+func TestDBAccountName_validation(t *testing.T) {
+	str := acctest.RandString(50)
+	cases := []struct {
+		Value    string
+		ErrCount int
+	}{
+		{
+			Value:    "ab",
+			ErrCount: 1,
+		},
+		{
+			Value:    "abc",
+			ErrCount: 0,
+		},
+		{
+			Value:    "cosmosDBAccount1",
+			ErrCount: 1,
+		},
+		{
+			Value:    "hello-world",
+			ErrCount: 0,
+		},
+		{
+			Value:    str,
+			ErrCount: 0,
+		},
+		{
+			Value:    str + "a",
+			ErrCount: 1,
+		},
+	}
+
+	for _, tc := range cases {
+		_, errors := validateDBAccountName(tc.Value, "azurerm_cosmosdb_account")
+
+		if len(errors) != tc.ErrCount {
+			t.Fatalf("Expected the AzureRM CosmosDB Name to trigger a validation error for '%s'", tc.Value)
+		}
+
+		_, errors = validateDBAccountName(tc.Value, "azurerm_sql_server")
+
+		if len(errors) != tc.ErrCount {
+			t.Fatalf("Expected the AzureRM SQL Server Name to trigger a validation error for '%s'", tc.Value)
+		}
```

Comment:
I think we can remove this secondary validation function?

Reply:
Can do!